### PR TITLE
feat(linter): `no-access-key` for eslint-plugin-jsx-a11y

### PR DESCRIPTION
**Pull request by yoshi2no**
partof: #1141

I've implemented `no-access-key` rule for jsx_a11y

originals:

- doc: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/no-access-key.md
- code: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/no-access-key.js
- test: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/__tests__/src/rules/no-access-key-test.js

### CODE REVIEW

File: crates/oxc_linter/src/snapshots/no_access_key.snap [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: no_access_key
+---
+  ⚠ eslint-plugin-jsx-a11y(no-access-key): No access key attribute allowed. Inconsistencies between keyboard shortcuts and keyboard commands used by screenreaders and keyboard-only users create
+  │ a11y complications.
+   ╭─[no_access_key.tsx:1:1]
+ 1 │ <div accesskey="h" />
```
Comment (camc314) [inline crates/oxc_linter/src/snapshots/no_access_key.snap:8]: shall we change this to report on the `accesskey` prop instead?

Comment (yoshi2no) [reply crates/oxc_linter/src/snapshots/no_access_key.snap:8]: @camc314 

I fixed this. 
Thank you for your review!

a76643009a0ae952b3e0402eef71d46e8ad24b44

File: crates/oxc_linter/src/snapshots/no_access_key.snap [hunk 2]
```diff
@@ -6,87 +6,87 @@ expression: no_access_key
   │ a11y complications.
    ╭─[no_access_key.tsx:1:1]
  1 │ <div accesskey="h" />
-   ·  ───
+   ·      ─────────────
    ╰────
   help: Remove the accessKey attribute.
 
   ⚠ eslint-plugin-jsx-a11y(no-access-key): No access key attribute allowed. Inconsistencies between keyboard shortcuts and keyboard commands used by screenreaders and keyboard-only users create
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/no_access_key.snap:13]: Should the second sentence go into `help` instead?

Comment (yoshi2no) [reply crates/oxc_linter/src/snapshots/no_access_key.snap]: @Boshen 

That would be nice. so I fixed it!
Thank you for your review!

**Comment by Boshen** [comment]
Thank you!

**Pull request merged**